lib/IRGen/IRGenDebugInfo.cpp

```diff
@@ -1946,7 +1946,7 @@ class IRGenDebugInfoImpl : public IRGenDebugInfo {
           /* DWARFAddressSpace */ std::nullopt, MangledName);
 
       // FIXME: Set DIFlagObjectPointer and make sure it is only set for `self`.
-      return PTy;
+      return DBuilder.createArtificialType(PTy);
```

Comment:
Sorry for the post-merge comment but I don't think we need to mark these artificial either tbh

There is no harm it leaving it off. The `artificial` flag shouldn't be driven by the `TypeKind`, but rather the variable being emitted (similar to the `DIFlagObjectPointer`)